astropy/io/ascii/tests/test_c_reader.py

```diff
@@ -1075,12 +1074,15 @@ def test_data_out_of_range(parallel, fast_reader, guess):
         elif TRAVIS:
             pytest.xfail("Multiprocessing can sometimes fail on Travis CI")
 
+    test_for_warnings = fast_reader and not parallel
+
     fields = ['10.1E+199', '3.14e+313', '2048e+306', '0.6E-325', '-2.e345']
     values = np.array([1.01e200, np.inf, np.inf, 0.0, -np.inf])
-    with catch_warnings(AstropyWarning) as w:
+    # NOTE: Warning behavior varies for the parameters being passed in.
+    with pytest.warns(None) as w:
```

Comment:
This test file is a tricky one. `nullcontext` doesn't work and too strict a `warns` catching doesn't work either; must be due to weird interactions between `pyx` and pure Python. 🤷 

At least this way, I *think* I didn't change the original intent of the tests.